src/lsstseries/analysis/structure_function/base_calculator.py

```diff
@@ -108,12 +107,25 @@ def _calculate_binned_statistics(self, statistic_to_apply="mean"):
             The first list contains the center of the delta_t bins.
             The second list contains the result of evaluating the
             statistic measure on the delta_flux values in each delta_t bin.
+
+        Notes
+        -----
+        1) Largely speaking this is a wrapper over Scipy's `binned_statistic`,
+        so any of the statistics supported by that function are valid inputs here.
+
+        2) It is expected that the shapes of `self._dts` and `sample_values` are
+        the same. Additionally, any entry at the i_th index of `self._dts` must
+        correspond to the same pair of observations as the entry at the i_th
+        index of `sample_values`.
         """
 
+        if sample_values is None:
+            sample_values = self._all_d_fluxes
+
```

Comment:
Is it worth checking that the sizes are the same here and raising an exception if they are not? I don't feel strongly either way.

Reply:
I've added a quick check just after this conditional, and I've added try/except blocks around the `binned_statistic` calls to handle exceptions from those due to incorrectly sized arrays.